### PR TITLE
[redis] adds additional sorted set type information

### PR DESCRIPTION
**Pull request by vishalkuo**
This adds type info for `zcount` and `zrank` to make further progress on #3544

### CODE REVIEW

File: third_party/2and3/redis/client.pyi [hunk 1]
```diff
@@ -242,14 +242,14 @@ class Redis(object):
         self, name: _Key, mapping: Mapping[_Key, _Str], nx: bool = ..., xx: bool = ..., ch: bool = ..., incr: bool = ...
     ) -> int: ...
     def zcard(self, name): ...
-    def zcount(self, name, min, max): ...
+    def zcount(self, name: _Key, min: _Str, max: _Str): ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi:245]: `min` and `max` can also be a `float` value. It returns an `int`.

File: third_party/2and3/redis/client.pyi [hunk 2]
```diff
@@ -242,14 +242,14 @@ class Redis(object):
...
     def zincrby(self, name, value, amount=...): ...
     def zinterstore(self, dest, keys, aggregate=...): ...
     def zlexcount(self, name, min, max): ...
     def zrange(self, name, start, end, desc=..., withscores=..., score_cast_func=...): ...
     def zrangebylex(self, name, min, max, start=..., num=...): ...
     def zrangebyscore(self, name, min, max, start=..., num=..., withscores=..., score_cast_func=...): ...
-    def zrank(self, name, value): ...
+    def zrank(self, name: _Key, value: _Key): ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi:252]: Returns an `Optional[int]`.

File: third_party/2and3/redis/client.pyi (hunk 1 above)
Comment (vishalkuo) [reply third_party/2and3/redis/client.pyi:245]: See the definition of `_Str`: `_Str = Union[bytes, float, int, Text]`. We can change the name of the variable to `_StrOrNumber` if you'd like?

Comment (srittau) [reply third_party/2and3/redis/client.pyi:245]: Sorry for missing that. We should maybe change it (maybe just to `_Value`?) at some point, but I believe it is out of scope for this PR.

Comment (vishalkuo) [reply third_party/2and3/redis/client.pyi]: I'll do it in another PR!

**Pull request merged**